source/logging.rst

```diff
@@ -0,0 +1,309 @@
+       were not specified at all.
+
+   * - Support for configuring where log messages should be output, including stderr,
+       stdout, and an output file with a configurable path.
+     - Support configuration via the environment variable ``MONGODB_LOGGING_PATH``.
+       
+       If the value is "stdout" or "stderr" (case-insensitive), log to the
+       corresponding output stream.
+       
+       Else, log to a file at the specified path. If the file already exists,
+       it MUST be appended to.
+
+       If the variable is not provided, log to stderr.
+
+   * - Support for configuring the maximum length for extended JSON documents
+       in log messages, with a default max length of 1000 characters.
+     - Support configuration via the environment variable
+       ``MONGODB_LOGGING_MAX_DOCUMENT_LENGTH``.
+      
+       When unspecified, any document longer than 1000 characters MUST be
+       truncated to 1000 characters.
+
+       When set to an integer value, any document longer than that value MUST
+       be truncated to that number of characters.
+
+       When set to the string "unlimited", documents MUST be included in full.
+
+       If the variable is set to an invalid value, it MUST be treated as if it
+       were not specified at all.
+
+Drivers MAY additionally provide support for enabling and configuring logging
+via API in a manner that does require code changes, and MAY support additional
+configuration options if it is idiomatic to do so.
+
+Components
+----------
+As noted above, drivers must support configuring logging verbosity on a
+per-component level. The below components currently exist and correspond to the
+listed specifications; this list is expected to grow over time.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Component Name
+     - Specification(s)
+     - Environment Variable
+
+   * - Command
+     - `Command Monitoring <../command-monitoring/command-monitoring.rst>`__
+     - ``MONGODB_LOGGING_COMMAND``
+
+   * - SDAM
+     - `Server Discovery and Monitoring
+       <../server-discovery-and-monitoring/server-discovery-and-monitoring.rst>`__
+     - ``MONGODB_LOGGING_SDAM``
+
+   * - Server Selection
+     - `Server Selection <../server-selection/server-selection.rst>`__
+     - ``MONGODB_LOGGING_SERVER_SELECTION``
+
+   * - Connection
+     - `Connection Monitoring and Pooling
+       <../connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst>`__
+     - ``MONGODB_LOGGING_CONNECTION``
+
+Severity Levels
+---------------
+Driver specifications defining log messages may use any of the following levels,
+inspired by the Syslog Protocol as described in `RFC 5424
+<https://tools.ietf.org/html/rfc5424>`__:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1 1
+
+   * - Code
+     - Level Name
+     - Meaning
+     - Environment Variable value (case-insensitive)
+
+   * - N/A
+     - Off
+     - Nothing is logged.
```

Comment:
I don't think `off` should be listed here, as it isn't a severity but a special configuration value. `off` may not be used when logging messages, but can be used by user to explicitly disable logging.

My suggestion would be to remove it here and add a paragraph on how to disable logging in the environment variable section above.